great_expectations/data_context/store/gx_cloud_store_backend.py

```diff
@@ -543,20 +543,48 @@ def remove_key(self, key):
             )
 
     def _update(self, key, value, **kwargs):
-        existing = self._get(key)
+        existing = self._get(key)["data"]
+
+        # if the provided key does not contain id (only name), cloud will return a list of resources filtered
+        # by name, with length >= 0, instead of a single object (or error if not found)
+        if isinstance(existing, list) and len(existing) > 1:
+            raise StoreBackendError(
+                f"Unable to update object in GX Cloud Store Backend: the provided key ({key}) maps "
+                f"to more than one object."
+            )
+        if isinstance(existing, list) and len(existing) == 0:
+            raise StoreBackendError(
+                f"Unable to update object in GX Cloud Store Backend: could not find object associated with key {key}."
+            )
+        if isinstance(existing, list) and len(existing) == 1:
+            existing = existing[0]
+
         if key[1] is None:
-            key = (key[0], existing["data"]["id"], key[2])
+            key = (key[0], existing["id"], key[2])
 
         return self.set(key=key, value=value, **kwargs)
 
     def _add_or_update(self, key, value, **kwargs):
         try:
-            existing = self._get(key)
+            existing = self._get(key)["data"]
         except StoreBackendError as e:
             logger.info(f"Could not find object associated with key {key}: {e}")
             existing = None
+
+        # if the provided key does not contain id (only name), cloud will return a list of resources filtered
+        # by name, with length >= 0, instead of a single object (or error if not found)
+        if isinstance(existing, list) and len(existing) > 1:
+            raise StoreBackendError(
+                f"Unable to update object in GX Cloud Store Backend: the provided key ({key}) maps "
+                f"to more than one object."
+            )
+        if isinstance(existing, list) and len(existing) == 0:
+            existing = None
+        if isinstance(existing, list) and len(existing) == 1:
+            existing = existing[0]
```

Comment:
Maybe we encapsulate this into a helper?

Reply:
Addressed in [d10e3e9](https://github.com/great-expectations/great_expectations/pull/8116/commits/d10e3e9cfa738f48386eb9d219460d6a518ea0de)